common/authorization/interceptor.go

```diff
@@ -45,42 +49,77 @@ func (a *interceptor) Interceptor(
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
 
-	var namespace string
-	requestWithNamespace, ok := req.(requestWithNamespace)
-	if ok {
-		namespace = requestWithNamespace.GetNamespace()
-	}
+	var claims *Claims
 
-	apiName := info.FullMethod
-	index := strings.LastIndex(apiName, "/")
-	if index > -1 {
-		apiName = apiName[index+1:]
+	if a.claimMapper != nil && a.authorizer != nil {
+		var tlsSubject *pkix.Name
+		var authHeaders []string
+
+		if md, ok := metadata.FromIncomingContext(ctx); ok {
+			authHeaders = md["authorization"]
+		}
+		if p, ok := peer.FromContext(ctx); ok {
+			if tlsAuth, ok := p.AuthInfo.(credentials.TLSInfo); ok {
+				if len(tlsAuth.State.VerifiedChains) > 0 && len(tlsAuth.State.VerifiedChains[0]) > 0 {
+					tlsSubject = &tlsAuth.State.VerifiedChains[0][0].Subject
```

Comment:
[Nit] can you add a comment as to why its [0][0] that matters and not [0][1] or [1][1] if those also existed?

Reply:
The assumption here is that we only expect a single verified chain of certs (first[0]). I don't know how we should handle a situation when more than one chain is presented which subject to use. I think for now it's okay for us to limit ourselves to one chain. We can always extend this logic later.

The second [0] is because we take the client cert (at the beginning of the chain), not intermediary CAs or the root CA (at the end of the chain). I think that is the correct logic because CAs are generally used for multiple subjects.

Reply:
This is great explanation. I beleive it should go to the comment.